pinot-common/src/main/java/org/apache/pinot/common/utils/fetcher/HttpSegmentFetcher.java

```diff
@@ -48,7 +50,10 @@ public void fetchSegmentToLocal(URI downloadURI, File dest)
     // Create a RoundRobinURIProvider to round robin IP addresses when retry uploading. Otherwise may always try to
     // download from a same broken host as: 1) DNS may not RR the IP addresses 2) OS cache the DNS resolution result.
     RoundRobinURIProvider uriProvider = new RoundRobinURIProvider(downloadURI);
-    int retryCount = Math.max(_retryCount, uriProvider.numAddresses());
```

Comment:
Let's use `Math.min()`.

We already read `_retryCount` in `BaseSegmentFetcher` with the following line:
```
    _retryCount = config.getProperty(RETRY_COUNT_CONFIG_KEY, DEFAULT_RETRY_COUNT);

```